File: chunk-optimizer-client/src/chunk_optimizer/exceptions.py
"""Exceptions for Chunk Optimizer Client"""


class ChunkOptimizerError(Exception):
    """Base exception for Chunk Optimizer errors"""
    pass


class AuthenticationError(ChunkOptimizerError):
    """Authentication failed"""
    pass


class RateLimitError(ChunkOptimizerError):
    """Rate limit exceeded"""
    pass


class NetworkError(ChunkOptimizerError):
    """Network error occurred"""
    pass


class ValidationError(ChunkOptimizerError):
    """Validation error"""
    pass

File: chunk-optimizer-js-client/src/exceptions.ts
export class ChunkOptimizerError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'ChunkOptimizerError';
  }
}

export class AuthenticationError extends ChunkOptimizerError {
  constructor(message: string = 'Authentication failed') {
    super(message);
    this.name = 'AuthenticationError';
  }
}

export class RateLimitError extends ChunkOptimizerError {
  constructor(message: string = 'Rate limit exceeded') {
    super(message);
    this.name = 'RateLimitError';
  }
}

export class NetworkError extends ChunkOptimizerError {
  constructor(message: string = 'Network error occurred') {
    super(message);
    this.name = 'NetworkError';
  }
}

export class ValidationError extends ChunkOptimizerError {
  constructor(message: string = 'Validation error') {
    super(message);
    this.name = 'ValidationError';
  }
}
